:py:mod:`flame.logging`
=======================

.. py:module:: flame.logging


Module Contents
---------------


Functions
~~~~~~~~~

.. autoapisummary::

   flame.logging.log_hook
   flame.logging.set_excepthook
   flame.logging.get_file_handler
   flame.logging.get_console_handler
   flame.logging.init_logger



Attributes
~~~~~~~~~~

.. autoapisummary::

   flame.logging._logger
   flame.logging.FILE_FORMAT
   flame.logging.CONSOLE_FORMAT


.. py:data:: _logger
   

   

.. py:data:: FILE_FORMAT
   :annotation: = %(asctime)s:%(levelname)-8s: %(message)s

   

.. py:data:: CONSOLE_FORMAT
   :annotation: = %(message)s

   

.. py:function:: log_hook(exc_type, exc_value, traceback)

   log the exception when it raise


.. py:function:: set_excepthook()


.. py:function:: get_file_handler(filename, fmt = FILE_FORMAT)


.. py:function:: get_console_handler()


.. py:function:: init_logger(rank = 0, filename = None, debug = False, force = False)

   :param rank: 目前只有rank0会输出到console和log file


